OpenRA.Mods.Common/SpriteLoaders/PngSheetLoader.cs

```diff
@@ -0,0 +1,135 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using System.IO;
+using OpenRA.FileFormats;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.FileFormats;
+
+namespace OpenRA.Mods.Common.SpriteLoaders
+{
+	public class PngSheetLoader : ISpriteLoader
+	{
+		class PngSheetFrame : ISpriteFrame
+		{
+			public Size Size { get; set; }
+			public Size FrameSize { get; set; }
+			public float2 Offset { get; set; }
+			public byte[] Data { get; set; }
+			public bool DisableExportPadding { get { return false; } }
+		}
+
+		public bool TryParseSprite(Stream s, out ISpriteFrame[] frames)
+		{
+			if (!Png.Verify(s))
+			{
+				frames = null;
+				return false;
+			}
+
+			var png = new Png(s);
+
+			var frameAmount = 0;
+			Rectangle[] frameRegions;
+			float2[] frameOffsets;
+
+			while (png.Meta.ContainsKey("Frame[" + frameAmount + "]"))
+				frameAmount++;
+
+			if (frameAmount > 0)
+			{
+				frameRegions = new Rectangle[frameAmount];
+				frameOffsets = new float2[frameAmount];
+
+				for (var i = 0; i < frameAmount; i++)
+				{
+					var coords = png.Meta["Frame[" + i + "]"].Split(',');
+
+					frameRegions[i] = new Rectangle(int.Parse(coords[0]), int.Parse(coords[1]), int.Parse(coords[2]), int.Parse(coords[3]));
```

Comment:
Style nit: It would be good for consistency and code reuse if we could use FieldLoader to parse these